models/src/main/protobuf/RhoTypes.proto

```diff
@@ -153,9 +153,8 @@ message New {
     Par p = 2 [(scalapb.field).no_box = true];
     // For normalization, uri-referenced variables come at the end, and in lexicographical order.
     repeated string uri = 3;
-    DeployId deployId = 4;
-    DeployerId deployerId = 5;
-    bytes locallyFree = 6 [(scalapb.field).type = "coop.rchain.models.AlwaysEqual[scala.collection.immutable.BitSet]"];
+    map<string, GUnforgeable> injections = 4;
```

Comment:
This should probably be just `map<string, Par>` though, as otherwise we're unable to treat `4` or `"woot"` as inputs. A bit of a bummer that GUnforgable aren't a variant of Expr - we could then just have `Expr` in here.

Reply:
> Isn't the idea that we want to inject values of types that can only be created by the system?

It's the other way around: we need to be able to inject values of types that can only be created by the system, among others - like e.g. `prechargeAmount(`sys:capser:prechargeAmount`)`, which would be a GInt.

Reply:
I see.

The disconnect was that I was thinking the idea is to create unforgeable ground types, e.g. `GUnforgeableInt`, and by that protecting us from users calling for example PoS methods.

Since this seems to not be the case: is there already a plan how we will prohibit user from calling the PoS contract?

Reply:
I think we should add a `validatorId: GDeployerId` parameter to each of the methods we only want the bonded validators to call. Based on that, we'd look if the corresponding publicKey is currently bonded, and short-circuit if not.

Notice the `bond` method should be available to user deploys, because it's merely a request to join the bonded validators group. The validator should probably provide its GDeployerId, so that we're sure the request comes from the interested party.